### PR TITLE
Remove hard coded DEFAULT_EXECUTION_BOUND as tests can provide their own gas budget parameters

### LINKED ISSUE #20: Allow rbpf tests to provide gas_budget as cfg
Some tests like test_set_bit_and_index_odd_size take way too long and fail with DEFAULT_EXECUTION_BOUND_SOLANA=10000 after adding a gas_limit #18. We should allow tests to provide gas_limit.

Something like:
```rs
     #[test]
+    #[gas_budget(compute_unit_limit=10000000)]
     fun test_set_bit_and_index_basic() {
         test_bitvector_set_unset_of_size(8)
     }
```

### PR DESCRIPTION
**Pull request by ksolana**
Follow up from #24

Fixes: #20

### CODE REVIEW

**Comment by ksolana** [comment]
One test fails, i'm investigating it.

**Comment by ksolana** [comment]
Fixed remaining tests.

File: external-crates/move/crates/move-compiler/src/unit_test/plan_builder.rs [hunk 1]
```diff
@@ -555,12 +550,12 @@ fn parse_gas_budget_attribute(context: &mut Context, gas_budget_attr: Option<&E:
                     TestingAttribute::GAS_BUDGET_HEAP_SIZE => {
                         let u = get_assigned_attribute_as_u64(context, TestingAttribute::GAS_BUDGET_HEAP_SIZE, attr_loc, attr)?;
                         // TODO: Do some sanity check that u shouldn't be larger than a max value.
-                        assigned_gas_budget.heap_size = u;
+                        assigned_gas_budget.heap_size = u as usize;
```
Comment (jcivlin) [inline external-crates/move/crates/move-compiler/src/unit_test/plan_builder.rs:553]: Here in each case we repeat a long constant name, I think Rust allows this syntax instead:
```Rust
                    x if x == TestingAttribute::GAS_BUDGET_HEAP_SIZE => {
                        let u = get_assigned_attribute_as_u64(context, x, attr_loc, attr)?;
```

File: external-crates/move/crates/move-compiler/src/unit_test/plan_builder.rs [hunk 2]
```diff
@@ -491,12 +489,9 @@ fn parse_failure_attribute(
...
 }
 
-fn parse_gas_budget_attribute(context: &mut Context, gas_budget_attr: Option<&E::Attribute>) -> Option<GasBudgetParams> {
-    let mut assigned_gas_budget = GasBudgetParams {
-        compute_budget:1,
-        heap_size: 1,
-        max_call_depth: 1,
-    };
+fn parse_gas_budget_attribute(context: &mut Context, gas_budget_attr: Option<&E::Attribute>) -> Option<SolanaGasBudgetParams> {
+
```
Comment (jcivlin) [inline external-crates/move/crates/move-compiler/src/unit_test/plan_builder.rs]: Empty line at the beginning of the function body, likely would be removed by an automatic formatter.

**Review by jcivlin — APPROVED** [review]
Suggesting some nit changes, ok to make fixes in consecutive PRs.

**Pull request merged**